R/helper_clean_model_data.R

```diff
@@ -97,6 +97,9 @@ clean_model_data <- function(formula,
 
   if(!missing(weights)){
     ret[["weights"]] <- model.extract(mf, "weights")
+    if (any(ret[["weights"]] <= 0)) {
+      stop("weights must all be positive and non-zero.")
```

Comment:
Zero weights should work?

Reply:
I agree, was in response to @jaspercooper's comments. I think I'll change to < 0.